src/model/track.rs

```diff
@@ -159,19 +161,105 @@ impl From<&SavedTrack> for Track {
 
 impl fmt::Display for Track {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(f, "{} - {}", self.artists.join(", "), self.title)
+        let active_fields = config::ACTIVE_FIELDS.read().unwrap().clone();
+        let mut fields: Vec<ActiveField> = Vec::new();
+        fields.append(
+            &mut active_fields
+                .left
+                .unwrap_or(vec![ActiveField::Artists, ActiveField::Title])
+                .clone(),
+        );
+        fields.append(
+            &mut active_fields
+                .center
+                .unwrap_or(vec![ActiveField::Album])
+                .clone(),
+        );
+        if fields.contains(&ActiveField::Title) && !fields.contains(&ActiveField::Artists) {
+            write!(f, "{}", self.title)
+        } else if fields.contains(&ActiveField::Artists) && !fields.contains(&ActiveField::Title) {
+            write!(f, "{}", self.artists.join(", "))
+        } else if !fields.contains(&ActiveField::Default)
+            && !fields.contains(&ActiveField::Title)
+            && !fields.contains(&ActiveField::Artists)
+        {
+            write!(f, "")
+        } else {
+            let mut title_index = 1;
+            let mut artists_index = 0;
+            for (i, field) in fields.iter().enumerate() {
+                match field {
+                    config::ActiveField::Title => title_index = i,
+                    config::ActiveField::Artists => artists_index = i,
+                    _ => {}
+                }
+            }
+            if title_index < artists_index {
+                write!(f, "{} - {}", self.title, self.artists.join(", "))
+            } else {
+                write!(f, "{} - {}", self.artists.join(", "), self.title)
+            }
```

Comment:
There is a lot of logic here to generate the output string for the columns. I'm wondering if there is a more dynamic approach, i.e. the user can provide something like `format_left = "%artists - %title"` which is then replaced.

Reply:
Ok, I like the idea. It would probably make the logic simpler. I'll see what I can do!